File: Design Strategies/greedy.java
import java.io.*;
import java.util.*;

public class greedy {
    static class Item {
        String name;
        int size;
        int value;

        Item(String name, int size, int value) {
            this.name = name;
            this.size = size;
            this.value = value;
        }
    }

    public static void main(String[] args) {
        Item[] items = {
                new Item("A", 10, 60),
                new Item("B", 20, 100),
                new Item("C", 30, 120),
                new Item("D", 5, 50)
        };
        int capacity = 20;

        // Sort items by value/size ratio in descending order
        Arrays.sort(items, (a, b) -> Double.compare(
                (double) b.value / b.size, (double) a.value / a.size));

        double totalValue = 0;
        int remainingCapacity = capacity;

        System.out.println("Selected items:");
        for (Item item : items) {
            if (item.size <= remainingCapacity) {
                System.out.println(item.name + " (size: " + item.size + ", value: " + item.value + ")");
                totalValue += item.value;
                remainingCapacity -= item.size;
            }
        }
        System.out.println("Total value: " + totalValue);
    }
}

File: Design Strategies/greedy.c
typedef struct {
    char name[10];
    int size;
    int value;
} Item;

int compare(const void* a, const void* b) {
    Item* itemA = (Item*)a;
    Item* itemB = (Item*)b;
    double ratioA = (double)itemA->value / itemA->size;
    double ratioB = (double)itemB->value / itemB->size;
    if (ratioA < ratioB) return 1;
    else if (ratioA > ratioB) return -1;
    else return 0;
}

int main() {
    Item items[] = {
        {"A", 10, 60},
        {"B", 20, 100},
        {"C", 30, 120},
        {"D", 5, 50}
    };
    int n = sizeof(items) / sizeof(items[0]);
    int capacity = 20;

    // Sort items by value/size ratio in descending order
    qsort(items, n, sizeof(Item), compare);

    double totalValue = 0;
    int remainingCapacity = capacity;

    printf("Selected items:\n");
    for (int i = 0; i < n; i++) {
        if (items[i].size <= remainingCapacity) {
            printf("%s (size: %d, value: %d)\n", items[i].name, items[i].size, items[i].value);
            totalValue += items[i].value;
            remainingCapacity -= items[i].size;
        }
    }
    printf("Total value: %.0f\n", totalValue);

    return 0;
}